test/pool/test_avg_pool2d_bwd.cpp

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2024, Advanced Micro Devices, Inc. All rights reserved.
+
+#include "gtest/gtest.h"
+#include "profiler/profile_avg_pool2d_bwd_impl.hpp"
+#include "test_pool_fwd_common.hpp"
+
+template <typename T>
+class AvgPool2dBWDTest : public ::testing::Test
+{
+    protected:
+    using InDataType      = std::tuple_element_t<0, T>;
+    using OutDataType     = std::tuple_element_t<1, T>;
+    using ComputeDataType = std::tuple_element_t<2, T>;
+
+    static std::vector<PoolingParam> params;
+
+    void Run()
+    {
+        for(auto param : this->params)
+        {
+            bool success = ck::profiler::
+                profile_avg_pool2d_bwd_impl<InDataType, OutDataType, ComputeDataType, NHWC, NHWC>(
+                    true,
+                    2,
+                    false,
+                    false,
+                    param.length_,
+                    param.window_spatial_lengths_,
+                    param.window_strides_,
+                    param.window_dilations_,
+                    param.input_left_pads_,
+                    param.input_right_pads_);
+            EXPECT_TRUE(success);
+        }
+    }
+};
+
+template <typename T>
+std::vector<PoolingParam> AvgPool2dBWDTest<T>::params = {
+    {{1, 1, 1, 1}, {1, 1}, {1, 1}, {1, 1}, {0, 0}, {0, 0}},
+    {{2, 16, 64, 64}, {64, 64}, {1, 1}, {1, 1}, {0, 0}, {0, 0}},
+    {{2, 16, 64, 64}, {4, 4}, {4, 4}, {2, 2}, {0, 0}, {0, 0}},
+    {{2, 32, 30, 30}, {2, 2}, {2, 2}, {1, 1}, {1, 1}, {1, 1}}};
+;
+
+using Avg_Pool_2D_f32_types        = ::testing::Types<std::tuple<F32, F32, F32>>;
+using Avg_Pool_2D_int8_types       = ::testing::Types<std::tuple<I8, I8, F32>>;
+using Avg_Pool_2D_f16_types        = ::testing::Types<std::tuple<F16, F16, F32>>;
+using Avg_Pool_2D_bf16_float_Types = ::testing::Types<std::tuple<BF16, BF16, F32>>;
+using Avg_Pool_2D_f8_float_Types   = ::testing::Types<std::tuple<F8, F8, F8>>;
```

Comment:
Please always do computation (here reduction in wider range data type, to prevent overflow)
Maybe that was the bug? 
```suggestion
using Avg_Pool_2D_f8_float_Types   = ::testing::Types<std::tuple<F8, F8, F32>>;
```